go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -235,6 +235,15 @@ func (bsnc *BaseSecondaryNetworkController) ensurePodForSecondaryNetwork(pod *ka
 		return err
 	}
 
+	if bsnc.IsPrimaryNetwork() && (bsnc.TopologyType() == "layer2" || bsnc.TopologyType() == "layer3") {
+		network := &nadapi.NetworkSelectionElement{
```

Comment:
@qinqon we'll need to come up w/ a way to figure out the name of the IPAMClaim for a persistent IPs user defined network. 

Maybe we can just hard-code it ?... Like ... `<vm name>.user-defined-primary-net`  

... or something like that. The kubevirt controller would create them w/ that name if there is a NAD on the namespace of the VM w/ the primary knob enabled.

Reply:
Let's just annotate the virt-launcher pods with ipamclaim name at kubevirt-ipam-controller `ovn.org/ipamclaim-refrence` or the like.

Reply:
why would we do that ? aren't we actually saying everywhere else that annotation / labeling based solutions are bad, and should not be used ?

Reply:
NetworkSelectionElement is an annotation too.

Reply:
This annotation is reconciled by kubevirt-ipam-controller if someone change it it will be restored

Reply:
an alternative is ovn-k reading the kubevirt vm label to compose the IPAMCLaim name.

Reply:
> NetworkSelectionElement is an annotation too.

It's part of a standard. I wouldn't put it in the same level. 

Can you first please explain to me why you're against just hard-coding this name ? It's by far the simplest solution from my perspective. 

Then we can evaluate alternative, sure.

> Let's just annotate the virt-launcher pods with ipamclaim name at kubevirt-ipam-controller k8s.ovn.org/ipamclaim-refrence or the like. 

I don't like this because it must happen in the webhook **and** the controller, if we want to put it back in case someone tampers with it.

> an alternative is ovn-k reading the kubevirt vm label to compose the IPAMCLaim name. 

I don't follow. Which label ?

Reply:
Thinking a bit more, I just realized that we need the annotation or label or whatever; otherwise, we'd do this always, depending on the network - and we don't want to do this for pods.